experimental/priority_sampler_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package experimental
+
+import (
+	"testing"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/jaeger-client-go"
+)
+
+func makePrioritySampler(t *testing.T) *PrioritySampler {
+	tagSampler := NewTagMatchingSampler("theWho", []TagMatcher{
+		{TagValue: "Bender", Firehose: false},
+		{TagValue: "Leela", Firehose: true},
+	})
+
+	return NewPrioritySampler(tagSampler, jaeger.NewConstSampler(false))
+}
+
+func TestPrioritySamplerShouldNotSampleOrFinalize(t *testing.T) {
+	tracer, closer := jaeger.NewTracer("svc", makePrioritySampler(t), jaeger.NewNullReporter())
+	defer closer.Close()
+
+	span := tracer.StartSpan("op1")
+	assert.False(t, span.Context().(jaeger.SpanContext).IsSampled())
+	assert.False(t, span.Context().(jaeger.SpanContext).IsSamplingFinalized())
+
+	span.SetOperationName("op2")
+	assert.False(t, span.Context().(jaeger.SpanContext).IsSampled())
+	assert.False(t, span.Context().(jaeger.SpanContext).IsSamplingFinalized())
+
+	span.Finish()
+	assert.False(t, span.Context().(jaeger.SpanContext).IsSampled())
+	assert.False(t, span.Context().(jaeger.SpanContext).IsSamplingFinalized())
+}
+
+func TestPrioritySampler(t *testing.T) {
```

Comment:
Perhaps something about tags?

Reply:
it's kind of the core test for the sampler, hence the top-level name